File: python.vb.ns.com/scripts/example/array.py
# coding: UTF-8
#
# 配列
#

print "---- 配列 ----"

### リスト
print "---- リスト ----"

arr = [200, 400, 100, 500]

print arr
print len(arr)
print arr[2]

# 値の変更
arr[2] = 90
print arr[2]

# 値の確認
print 400 in arr

# ソート
arr.sort()
print arr
arr.reverse()
print arr

# 連結
arr = ["a", "b", "c"]
print ".".join(arr)

# 連続データの作成
print range(10)
print range(3, 10)
print range(3, 10, 2)

### タプル
# 要素の変更ができない配列
print "---- タプル ----"

arr = (100, 30, 80, 200)

print arr
print len(arr)
print arr * 2

# arr[2] = 10 # TypeError: 'tuple' object does not support item assignment

### セット（集合型）
# 要素の重複を許さない
print "---- セット ----"

arr1 = set([1, 2, 3, 4, 3, 1, 2]) # 3, 1, 2 は無視される
arr2 = set([3, 4, 5, 6])

print arr1

# 演算
print arr1 - arr2 # 差集合：arr1にあってarr2にないもの
print arr1 | arr2 # 和集合：両方の値すべて
print arr1 & arr2 # 積集合：重複したものを表示
print arr1 ^ arr2 # どちらかにしかない

File: java.vb.ns.com/scripts/example/Array.java
//
// 配列
//
public class Array {
    public static void main(String[] args) {
        System.out.println("---- array ----");

        // [] をつけることで配列であることを宣言
        int arr[] = new int[4]; // 領域の確保

        System.out.println(arr.length);

        arr[0] = 100;
        arr[1] = 200;
        arr[2] = 400;
        arr[3] = 50;

        // 直接代入
        int foo[] = {10, 20, 50, 30, 8};

        System.out.println(foo.length);
    }
}
